src/pages/modules/ModulesPage.slice.ts

```diff
@@ -0,0 +1,138 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import {
+  IModuleState,
+  IModulesGETFailure,
+  IModulesPUT,
+  IModulesPUTFailure,
+  IModule,
+  AlertType,
+} from "../../shared/types";
+import { AlertMsg } from "./config";
+
+const initialModuleState: IModuleState = {
+  modules: [],
+  isLoading: true,
+  feedback: {
+    message: "",
+    display: false,
+    type: AlertType.info,
+  },
+};
+
+export function getInitialModuleState(): IModuleState {
+  return { ...initialModuleState };
+}
+
+export const moduleSlice = createSlice({
+  name: "modules",
+  initialState: getInitialModuleState(),
+  reducers: {
+    /* GET Request Modules */
+    requestModules: (state, action: PayloadAction<void>) => {
+      return { ...state, isLoading: true };
+    },
+
+    // action.type = "modules/requestModulesSuccess"
+    requestModulesSuccess: (state, action: PayloadAction<IModule[]>) => {
+      console.log("getting modules success");
```

Comment:
nit: console.log

Reply:
I will remove it with the commit for test cases